server/datastore/mysql/migrations/tables/20220802135510_AddMobileDeviceManagementTable.go

```diff
@@ -0,0 +1,66 @@
+package tables
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20220802135510, Down_20220802135510)
+}
+
+func Up_20220802135510(tx *sql.Tx) error {
+	logger.Info.Println("Creating table mobile_device_management_solutions...")
+
+	// An mdm ID identifies a Vendor + Server URL combination (e.g. Jamf + https://company.jamfcloud.com)
+	// A distinct server URL for the same vendor results in different MDM ID.
+	_, err := tx.Exec(`
+CREATE TABLE mobile_device_management_solutions (
+  id            INT(10) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  name          VARCHAR(100) NOT NULL,
+  server_url    VARCHAR(255) NOT NULL,
+  created_at    TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at    TIMESTAMP NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+
+  UNIQUE KEY idx_mobile_device_management_solutions_name (name, server_url)
+)`)
+	if err != nil {
+		return errors.Wrapf(err, "create table")
+	}
+	logger.Info.Println("Done creating table mobile_device_management_solutions...")
+
+	logger.Info.Println("Adding column mdm_id to table host_mdm...")
+	// adding as NULLable to prevent costly migration for users with many hosts,
+	// the mdm_id will be lazily populated as MDM query results get returned by
+	// hosts.
+	_, err = tx.Exec(`ALTER TABLE host_mdm ADD COLUMN mdm_id INT(10) UNSIGNED NULL;`)
```

Comment:
Would it be appropriate to use an an online DDL operation here? Don't know how big the `host_mdm` table is.

Reply:
AFAIK we haven't started deploying live migrations yet (nor implemented, I believe), so this would be applied while fleet is not running.

Reply:
Sounds good, thanks